File: notiflut_daemon/native/notification_server/src/theme.rs
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize)]
struct Color(u32);

#[derive(Deserialize, Serialize)]
struct Radius(u32);

#[derive(Deserialize, Serialize)]
struct NotificationStyle {
    background_color: Color,
    border_radius: Radius,
    text_color: Color,
}

#[derive(Deserialize, Serialize)]
struct NotificationCenterStyle {
    background_color: Color,
}

#[derive(Deserialize, Serialize)]
struct PopupStyle {
    background_color: Color,
}

#[derive(Deserialize, Serialize)]
struct Theme {
    notification_center: NotificationCenterStyle,
    popup: PopupStyle,
    notification: NotificationStyle,
}

File: notiflut_daemon/lib/theme.dart
import 'package:flutter/material.dart';

class NotificationStyle {
  Color backgroundColor;
  Radius borderRadius;

  NotificationStyle({
    required this.backgroundColor,
    required this.borderRadius,
  });
}

class NotificationCenter {
  Color backgroundColor;

  NotificationCenter({
    required this.backgroundColor,
  });
}

class Theme {
  NotificationStyle notificationStyle;
  Theme({
    required this.notificationStyle,
  });
}

class Style {
  Theme light;
  Theme dark;
  Style({
    required this.light,
    required this.dark,
  });
}
